src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -22,4 +22,33 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object rectangle) {
+        if (rectangle == this) {
+            return true;
+        }
+        if (rectangle == null) {
+            return false;
+        }
+        if (!rectangle.getClass().equals(Rectangle.class)) {
+            return false;
+        }
+        Rectangle castedRectangle = (Rectangle) rectangle;
+        return ((color == null && castedRectangle.color == null)
+                || (color != null && color.equals(castedRectangle.color)))
+                && ((width == null && castedRectangle.width == null)
+                || (width != null && width.equals(castedRectangle.width)))
+                && ((length == null && castedRectangle.length == null)
+                || (length != null && length.equals(castedRectangle.length)));
```

Comment:
good job!
this may be simplified a bit

```suggestion
        return (color == castedRectangle.color
                || (color != null && color.equals(castedRectangle.color)))
                && (width == castedRectangle.width
                || (width != null && width.equals(castedRectangle.width)))
                && (length == castedRectangle.length
                || (length != null && length.equals(castedRectangle.length)));
```